### PR TITLE
Add categories to metadata

### PR DESCRIPTION
**Pull request by Chuxel**
Also includes (Community) in descriptions for those not created by MS or core extension engineering teams.

### CODE REVIEW

**Review by 2percentsilk — APPROVED** [review]

**Comment by 2percentsilk** [comment]
@Chuxel this looks good! Do we have anything we're waiting on for merging or can I go ahead?

**Review by bamurtaugh — APPROVED** [review]
LGTM - thanks for adding these changes! I think adding the "Community" signifiers at a top-level makes sense. I also see based on the template that any new definitions would by default be Community but cannot be Core.

File: container-templates/docker-compose/README.md [hunk 1]
```diff
@@ -1,17 +1,18 @@
...
+*[A short description of the the purpose of the definition goes here.]*
+
+| Metadata                    | Value                                                                        |
+|---------------------------- | -----------------------------------------------------------------------------|
+| *Contributors*              | [Your name, GitHub profile]                                                  |
+| *Categories*                | Community, [Languages, Frameworks, Services, Azure, GCP, AWS, GitHub, Other] |
+| *Definition type*           | Docker Compose                                                               |
+| *Works in Codespaces*       | Yes / No                                                                     |
+| *Container host OS support* | Linux, macOS, Windows                                                        |
+| *Container OS*              | [OS used by continer - e.g. Debian]                                          |
```
Comment (bamurtaugh) [inline container-templates/docker-compose/README.md:14]:
```suggestion
| *Container OS*              | [OS used by container - e.g. Debian]                                          |
```

File: container-templates/dockerfile/README.md [hunk 2]
```diff
@@ -1,18 +1,18 @@
...
-| *Container OS*              | [OS used by continer - e.g. Debian]              |
-| *Languages, platforms*      | [Languages and platforms the container supports] |
+| Metadata                    | Value                                                                        |
+|---------------------------- | -----------------------------------------------------------------------------|
+| *Contributors*              | [Your name, GitHub profile]                                                  |
+| *Categories*                | Community, [Languages, Frameworks, Services, Azure, GCP, AWS, GitHub, Other] |
+| *Definition type*           | Dockerfile                                                                   |
+| *Works in Codespaces*       | Yes / No                                                                     |
+| *Container host OS support* | Linux, macOS, Windows                                                        |
+| *Container OS*              | [OS used by continer - e.g. Debian]                                          |
```
Comment (bamurtaugh) [inline container-templates/dockerfile/README.md:14]:
```suggestion
| *Container OS*              | [OS used by container - e.g. Debian]                                          |
```

File: container-templates/image/README.md [hunk 3]
```diff
@@ -1,17 +1,18 @@
...
+*[A short description of the the purpose of the definition goes here.]*
+
+| Metadata                    | Value                                                                        |
+|---------------------------- | -----------------------------------------------------------------------------|
+| *Contributors*              | [Your name, GitHub profile]                                                  |
+| *Categories*                | Community, [Languages, Frameworks, Services, Azure, GCP, AWS, GitHub, Other] |
+| *Definition type*           | Image                                                                        |
+| *Works in Codespaces*       | Yes / No                                                                     |
+| *Container host OS support* | Linux, macOS, Windows                                                        |
+| *Container OS*              | [OS used by continer - e.g. Debian]                                          |
```
Comment (bamurtaugh) [inline container-templates/image/README.md:14]:
```suggestion
| *Container OS*              | [OS used by container - e.g. Debian]                                          |
```

**Review by egamma — APPROVED** [review]

**Comment by egamma** [comment]
@Chuxel LGTM. 

Just an observation, there are some gaps/inconsistencies when it comes to DB coverage.

mongo:
- javascript-node-mongo - category: core
- typescript-node-mongo - _does not exist_

postgres:
- python-3-postgres - category: core
- javascript-node-postgres - category: community
- typescript-node-postgres - _does not exist_

**Comment by Chuxel** [comment]
> Just an observation, there are some gaps/inconsistencies when it comes to DB coverage.
>
> mongo:
> 
> javascript-node-mongo - category: core
> typescript-node-mongo - does not exist
>
> postgres:
>
> python-3-postgres - category: core
> javascript-node-postgres - category: community
> typescript-node-postgres - does not exist

@egamma Yeah `javascript-node-postgres` was contributed by the community, so that is accurate. We could consider "adopting" it.  For the typescript ones, I'm currently hesitant to add additional non-community maintained definitions given the volume we have already. Let's discuss.

**Pull request merged**